core/templates/components/forms/schema-viewers/schema-based-custom-viewer.directive.spec.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for Schema Based Custom Viewer Directive
+ */
+
+import { importAllAngularServices } from 'tests/unit-test-utils.ajs';
+
+describe('Schema Based Custom Viewer Directive', () => {
+  let ctrl = null;
+  let $rootScope = null;
+  let $scope = null;
+  let directive = null;
+
+  beforeEach(angular.mock.module('oppia'));
+  importAllAngularServices();
+
+  beforeEach(angular.mock.inject(function($injector) {
+    $rootScope = $injector.get('$rootScope');
+    $scope = $rootScope.$new();
+
+    directive = $injector.get('schemaBasedCustomViewerDirective')[0];
+    ctrl = $injector.instantiate(directive.controller, {
+      $scope: $scope
+    });
+  }));
+
+  it('should initialize directive', () => {
+    expect(ctrl).toBeDefined();
+  });
```

Comment:
Is it possible to make this test more thorough? Right now it doesn't seem like we're asserting very much. Here's an example of a more comprehensive set of tests for a directive:

https://github.com/oppia/oppia/blob/develop/core/templates/components/forms/custom-forms-directives/apply-validation.directive.spec.ts

Reply:
Thanks @U8NWXD. The directives do not have any more attributes or defined functions that could be tested using the spec file. I followed the example of [schema-based-html-viewer.directive.spec.ts](https://github.com/oppia/oppia/blob/develop/core/templates/components/forms/schema-viewers/schema-based-html-viewer.directive.spec.ts) present in the same directory.